Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -4750,6 +4750,10 @@ - (void)_keyboardDidRequestDismissal:(NSNotification *)notification
 {
     if (_isEditable && [self isFirstResponder])
         _keyboardDidRequestDismissal = YES;
+
+#if USE(UICONTEXTMENU)
+    [_fileUploadPanel repositionContextMenuIfNeeded:YES];
```

Comment:
Nit - for a bit more clarity, you could do one of the following:

1. Make the `BOOL` argument an `enum class` instead (e.g. `enum class KeyboardIsDismissing : bool { No, Yes };`), or

2. Rename the method to something along the lines of `- (void)repositionContextMenuAfterKeyboardDismissal:(BOOL)isDismissing;`

Reply:
I can do #2 but what is the idea behind #1 ? Why would it give more clarity?

Reply:
Correct, otherwise it's unclear what the boolean actually represents.

For example, it would read like `repositionContextMenuIfNeededAfterKeyboardDismissal:NO`, which is confusing because that call could still reposition the menu.

Reply:
Right — the root problem is that from just looking at the call site (`[_fileUploadPanel repositionContextMenuIfNeeded:YES];`), it's not obvious what `YES` represents.

(1) and (2) both help to mitigate this by adding more context to clarify that `YES` in this case refers to the fact that it denotes whether the call is after keyboard dismissal.

Reply:
Since repositioning the menu is needed before keyboard dismissal, I'm thinking that the function name shouldn't change now... but (1) definitely makes sense. Smth like `[_fileUploadPanel repositionContextMenuIfNeeded:KeyboardIsDismissing::YES]` which would signify reposition if needed AND given keyboard is dismissing rn?

Reply:
Yeah, that seems fine.